File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    answer = 0
    memo = [0, 1, 1] + [0] * n
    for i in range(3, n+2):
        memo[i] = memo[i-1] + memo[i-2]
    answer = memo[n] % 1234567
    return answer

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    var answer = 0;
    let memo = new Array(n+1);
    
    memo[0] = 0;
    memo[1] = 1;
    memo[2] = 1;

    for (let i=3; i<n+2; i++) {
        memo[i] = (memo[i-1] + memo[i-2]) % 1234567
    } 
    answer = memo[n]
    return answer;
}
